From 8fe2ac644e37de31b8780f2f38499bef82821003 Mon Sep 17 00:00:00 2001
From: Michel Alexandre Salim <salimma@fedoraproject.org>
Date: Tue, 2 Aug 2011 20:04:44 +0200
Subject: [PATCH] Use provided CFLAGS when building miro-segmenter

miro_segmenter is built not as a Python extension module, but as a
normal C program, by directly invoking the system's C compiler.

As such, no compiler flag is passed to the compiler, resulting in an
unoptimized binary. This patch passes the value of the CFLAGS
environment variable, if set, to the compiler.

Signed-off-by: Michel Alexandre Salim <salimma@fedoraproject.org>
---
 tv/linux/setup.py |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/tv/linux/setup.py b/tv/linux/setup.py
index 54ba112..923166e 100644
--- a/tv/linux/setup.py
+++ b/tv/linux/setup.py
@@ -436,7 +436,8 @@ class build(distutils.command.build.build):
         cc.add_include_dir('/usr/include/ffmpeg')
         output_dir = os.path.join(self.build_base, 'miro-segmenter')
         segmenter_objs = cc.compile([segmenter_src],
-                                    output_dir=output_dir)
+                                    output_dir=output_dir,
+                                    extra_preargs=(os.environ.get("CFLAGS") or "").split())
         cc.link_executable(segmenter_objs, 'miro-segmenter',
                            output_dir=output_dir)
         segmenter_exe = os.path.join(output_dir, 'miro-segmenter')
-- 
1.7.6

